libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs

```diff
@@ -629,9 +633,19 @@ private static LuisRecognizerOptions BuildLuisRecognizerOptionsV2(LuisApplicatio
         private async Task<RecognizerResult> RecognizeInternalAsync(ITurnContext turnContext, LuisRecognizerOptions predictionOptions, Dictionary<string, string> telemetryProperties, Dictionary<string, double> telemetryMetrics, CancellationToken cancellationToken)
         {
             var recognizer = predictionOptions ?? _luisRecognizerOptions;
-            var result = await recognizer.RecognizeInternalAsync(turnContext, HttpClient, cancellationToken).ConfigureAwait(false);
-            await OnRecognizerResultAsync(result, turnContext, telemetryProperties, telemetryMetrics, cancellationToken).ConfigureAwait(false);
-            return result;
+            var cached = turnContext.TurnState.Get<RecognizerResult>(_cacheKey);
+            if (cached == null)
+            {
+                var result = await recognizer.RecognizeInternalAsync(turnContext, HttpClient, cancellationToken).ConfigureAwait(false);
+                await OnRecognizerResultAsync(result, turnContext, telemetryProperties, telemetryMetrics, cancellationToken).ConfigureAwait(false);
+                turnContext.TurnState.Set(_cacheKey, result);
+                _luisRecognizerOptions.TelemetryClient.TrackEvent("Luis result cached", telemetryProperties, telemetryMetrics);
```

Comment:
What is the scenario where just caching the most recent value is useful?

Reply:
https://github.com/microsoft/botbuilder-dotnet/issues/5593#issuecomment-861088870
As Steve mentioned in the issue comment, the twice Luis api calls happened consecutively in the same turn. One is for evaluation of `allowInterruption`, the other one is for recognize the intent in the child dialog. 
Therefore, if we cached the most recent result of luis recognizer, we don't need another request for the next call in the same turn for the same utterance.

Reply:
My concern is that this is a fix that is specific to LUIS and it implies that all recognizers should be prepared for multiple calls of the same utterance.  Seems like a better fix would either not to do multiple calls or cache somewhere else.  I'll chat to Steve and Tom and see what they think tomorrow.